app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2100,7 +2103,16 @@ class BrowserTabFragment :
             configureWebViewForAutofill(it)
             printInjector.addJsInterface(it) { viewModel.printFromWebView() }
             autoconsent.addJsInterface(it, autoconsentCallback)
-            contentScopeScripts.addJsInterface(it)
+            contentScopeScripts.register(
+                it,
+                object : JsMessageCallback(this) {
+                    override fun process(method: String) {
+                        runCatching {
+                            callback.javaClass.getDeclaredMethod(method)
+                        }.getOrNull()?.invoke(callback)
+                    }
+                },
```

Comment:
Is there a cleaner way of doing this? It doesn't seem like something a consumer of this method should be worrying about?

Reply:
This is not really used yet so I could remove it. We are using reflection to invoke the callback so we can have a single callback interface and let each handler call whichever methods they need. The idea is that this code should never change. You can see how this is used in `SubscriptionMessagingInterface`